adcm_aio_client/core/objects/cm.py

```diff
@@ -117,13 +113,51 @@ class ServicesNode(PaginatedChildAccessor[Cluster, Service, None]):
 
 
 class Component(InteractiveChildObject[Service]):
-    @property
-    def id(self: Self) -> int:
-        return int(self._data["id"])
 
     def get_own_path(self: Self) -> Endpoint:
         return (*self._parent.get_own_path(), "components", self.id)
 
 
 class ComponentsNode(NonPaginatedChildAccessor[Service, Component, None]):
     class_type = Component
+
+
+class HostProvidersNode(PaginatedChildAccessor): ...
+
+
+class Host(Deletable, InteractiveChildObject):
+
+    @property
+    def name(self: Self) -> str:
+        return str(self._data["name"])
+
+    @property
+    def description(self: Self) -> str:
+        return str(self._data["description"])
+
+    async def get_status(self: Self) -> ADCMEntityStatus:
+        response = await self._requester.get(*self.get_own_path())
+        return ADCMEntityStatus(response.as_dict()["status"])
+
+    @cached_property
+    def cluster(self: Self) -> Cluster | None:
+        if not self._data["cluster"]:
+            return None
+        return Cluster(requester=self._requester, data=self._data["cluster"])
```

Comment:
`cluster` info is not enough here, should be re-requested by cluster id, same for hostprovider

Reply:
Let's expand `RootInteractiveObject`:

```python
# we have this
class RootInteractiveObject(InteractiveObject):
    PATH_PREFIX: str

    def get_own_path(self: Self) -> Endpoint:
        # change here
        return self._build_own_path(self.id)

    # let's add this one
    @classmethod
    async def with_id(cls: type[Self], requester: Requester, id: int) -> Self:
        object_path = cls._build_own_path(id)
        response = await requester.get(*object_path)
        return cls(requester=requester, data=response.as_dict()) 

    @classmethod
    def _build_own_path(cls, id: int) -> Endpoint:
        return cls.PATH_PREFIX, id

# and now we can use it in methods like cluster

class Host:
    ...
    @cached_property
    async def cluster(self) -> Cluster | None:
        if notindata:
            return None

        return await Cluster.with_id(requester=self._requester, id=self._data["cluster"]["id"]
```

Reply:
done